coderd/database/dbmem/dbmem.go

```diff
@@ -5896,14 +5896,24 @@ func (q *FakeQuerier) GetWorkspaceAgentScriptTimingsByBuildID(ctx context.Contex
 			}
 		}
 
+		var agent database.WorkspaceAgent
+		for _, a := range agents {
+			if a.ID == script.WorkspaceAgentID {
+				agent = a
+				break
```

Comment:
What if no agent is found?

Reply:
In this case, it is not possible to happen since a `WorkspaceAgentScript` always has an agent.

Reply:
Right, but we should still account for this in code (i.e. raise an error if we encounter this unexpected scenario).